src/error.ts

```diff
@@ -170,6 +170,27 @@ export class MongoDriverError extends MongoError {
   }
 }
 
+/**
+ * An error generated when the driver encounters unexpected unexpected input
+ * or reaches an unexpected/invalid internal state
+ *
+ * @privateRemarks
+ * Should **never** be directly instantiated.
+ *
+ * @public
+ * @category Error
+ */
+export class MongoRuntimeError extends MongoDriverError {
+  code?: string;
```

Comment:
MongoDriverError already has the code declared, so I don't think it's necessary to redeclare it on the subclass unless we're intending to remove it from the parent class; I also want to note that the code for MongoDriverError is used for exactly one error and I think there is a ticket that is meant to find a proper way to deal with that case (the ENOENT case); for now, I'd say just leave the explicit declaration off this class

Reply:
(You could also consider making the actual codes a string enum for the more specific, directly instantiated, classes down the line)

Reply:
Just removed the code from this class because it was redundant like you said.